core/src/main/java/org/pentaho/di/connections/vfs/provider/ConnectionFileSystem.java

```diff
@@ -31,93 +33,170 @@
 import org.apache.commons.vfs2.provider.AbstractFileName;
 import org.apache.commons.vfs2.provider.AbstractFileObject;
 import org.apache.commons.vfs2.provider.AbstractFileSystem;
-import org.pentaho.di.connections.ConnectionDetails;
 import org.pentaho.di.connections.ConnectionManager;
 import org.pentaho.di.connections.vfs.VFSConnectionDetails;
+import org.pentaho.di.connections.vfs.VFSConnectionManagerHelper;
 import org.pentaho.di.core.bowl.Bowl;
+import org.pentaho.di.core.exception.KettleException;
 import org.pentaho.di.core.variables.VariableSpace;
 import org.pentaho.di.core.variables.Variables;
+import org.pentaho.di.core.vfs.IKettleVFS;
 import org.pentaho.di.core.vfs.KettleVFS;
+import org.pentaho.di.core.vfs.KettleVFSFileSystemException;
 import org.pentaho.di.core.vfs.configuration.IKettleFileSystemConfigBuilder;
 import org.pentaho.di.core.vfs.configuration.KettleFileSystemConfigBuilderFactory;
 import org.pentaho.di.core.vfs.configuration.KettleGenericFileSystemConfigBuilder;
 
+import java.io.IOException;
 import java.util.Collection;
+import java.util.Objects;
 
 public class ConnectionFileSystem extends AbstractFileSystem implements FileSystem {
 
   public static final String CONNECTION = "connection";
-  public static final String DOMAIN_ROOT = "[\\w]+://";
 
-  public ConnectionFileSystem( FileName rootName, FileSystemOptions fileSystemOptions ) {
+  @NonNull
+  private final ConnectionManager connectionManager;
+
+  @NonNull
+  private final VFSConnectionManagerHelper vfsConnectionManagerHelper;
+
+  public ConnectionFileSystem( @NonNull FileName rootName,
+                               @NonNull FileSystemOptions fileSystemOptions,
+                               @NonNull ConnectionManager connectionManager,
+                               @NonNull VFSConnectionManagerHelper vfsConnectionManagerHelper ) {
     super( rootName, null, fileSystemOptions );
+
+    this.connectionManager = Objects.requireNonNull( connectionManager );
+    this.vfsConnectionManagerHelper = Objects.requireNonNull( vfsConnectionManagerHelper );
   }
 
-  /**
```

Comment:
Moved to `createFile( AbstractFileName )` and `VFSConnectionFileNameTransformer#toProviderUri()`.